docs/generated/go.md

```diff
@@ -0,0 +1,201 @@
+---
+bodyclass: docs
+headline: Go Generated Code Guide
+layout: docs
+sidenav: doc-side-quickstart-nav.html
```

Comment:
I'd also maybe put this in docs/reference rather than docs/generated, it means we have a section for various reference docs that might come up *including* generated code (eg in the current protocol buffers docs, the reference section includes proto language specs as well as generated code guides and API docs)